docs/_source/guides/train_a_model.ipynb

```diff
@@ -0,0 +1,358 @@
+    "import argilla as rg\n",
+    "\n",
+    "dataset_rg = rg.load(\"<my_dataset>\")\n",
+    "dataset_rg.prepare_for_training(framework=\"spark-nlp\", train_size=1)\n",
+    "# <pd.DataFrame>\n",
+    "```\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "99a0a8a2",
+   "metadata": {},
+   "source": [
+    "## Train directly\n",
+    "\n",
+    "The `ArgillaTrainer` is an wrapper around many of our favorite NLP libraries, ad provides a very intuitive abstract workflow to facilitate simple training workflows using decent default pre-set configuration without having to worry about any data transformations from Argilla. We plan on adding more support for tasks and frameworks like OpenAI, AutoTrain and SageMaker during coming releases.\n",
+    "\n",
+    "| Framework/Task    | TextClassification | TokenClassification | Text2Text |\n",
+    "|-------------------|--------------------|---------------------|-----------|\n",
+    "| Transformers      | ✔️                  | ✔️                   |           |\n",
+    "| spaCy             | ✔️                  | ✔️                   |           |\n",
+    "| SetFit            | ✔️                  |                     |           |"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "6b318950",
+   "metadata": {},
+   "source": [
+    "### The `ArgillaTrainer`\n",
+    "\n",
+    "We can use the `ArgillaTrainer` to train directly for frameworks by `spacy`, `setfit` and `transformers` as framework variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0dfd78f4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import argilla as rg\n",
+    "from argilla.training import ArgillaTrainer\n",
+    "\n",
+    "trainer = ArgillaTrainer(\n",
+    "    name=\"<my_dataset_name>\",\n",
+    "    workspace=\"<my_workspace_name>\",\n",
+    "    framework=\"<my_framework>\",\n",
+    "    train_size=0.8\n",
+    ")\n",
+    "\n",
+    "trainer.train(path=\"<my_model_path>\")\n",
+    "records = trainer.predict(\"The ArgillaTrainer is great!\", as_argilla_records=True)\n",
+    "rg.log(records=records, name=\"<my_dataset_name>\", workspace=\"<my_workspace_name>\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "84725635",
+   "metadata": {},
+   "source": [
+    "### Update training config\n",
+    "\n",
+    "The trainer also has an `ArgillaTrainer.update_config()` method, which maps `**kwrags` to the respective framework. So, these can be derived from the underlying framework that was used to initialize the trainer. Underneath, an overview of these variables is given for the supported frameworks. Note that, you don't need to pass all of them directly and that the values below are their default configurations."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "9cfa853d",
+   "metadata": {},
+   "source": [
+    "#### SetFit\n",
```

Comment:
Something went wrong with the code highlighting here: https://docs.argilla.io/en/feat-2504-add-an-argillatraining-module-to-the-core-library-with-training-workflows/guides/train_a_model.html#id1